### ISSUE TITLE
Toolbar buttons Works on iPad but not on iPhone.

### ISSUE THREAD

**Issue by xiaoxidong** [Heming-Zhong/marknote#2]

**Comment by Heming-Zhong** [comment]
Yes It's a bug. The native Toolbar buttons is implemented by calling `click` action of editor's own toolbar button which is hidden. However, in a mobile browser, the corresponding action is a custom `touchstart` event. Here is a related issue: Vanessa219/vditor/issues/321 .

To switch to iPhone: change all `Setxxx` functions in  marknote/Editor/Editor.bundle/Vditor/src/test.html to:

```javascript
Setxxx = function() {
    <EditorToolbarButtonElement>.dispatchEvent(new CustomEvent("touchstart"))
}
```

For example, change `SetUnorderedList` function to:

```javascript
// original 
SetUnorderedList = function() {
    editor.vditor.toolbar.elements.list.firstElementChild.click()
}

// new
SetUnorderedList = function() {
               editor.vditor.toolbar.elements.list.firstElementChild.dispatchEvent(new CustomEvent("touchstart"))
}
```

However, After these changes, the toolbar buttons will not work on iPad. In order to make these buttons on both devices, code needed to identify the device class or UA class.
